lib/kinu/configuration.rb

```diff
@@ -1,11 +1,15 @@
 module Kinu
   class Configuration
-    attr_writer :host, :port, :ssl
+    attr_writer :host, :upload_host, :port, :ssl
 
     def host
       @host || ""
     end
 
+    def upload_host
+      @upload_host || host || ""
```

Comment:
IMO

```
def upload_host
  @upload_host || nil
end
```

I think the Configuration class will be simpler to return the setting directly